volatility3/framework/layers/intel.py

```diff
@@ -186,18 +186,45 @@ def _translate_entry(self, offset: int) -> Tuple[int, int]:
 
         Returns the translated entry value
         """
+        offset &= self.address_mask
+
+        if not (self.minimum_address <= offset <= self.maximum_address):
+            raise exceptions.InvalidAddressException(
+                offset, f"Address {offset:#x} outside virtual address range"
+            )
+
+        page_address = offset & self.page_mask
+        return self._translate_page(page_address)
+
+    @functools.lru_cache(maxsize=1024)
+    def _translate_page(self, page_address: int) -> int:
+        """Translates a page address based on paging tables.
+
+        Args:
+            page_address: The page base address
+
+        Returns:
+            the translated entry value
+        """
+        if page_address & ~self.page_mask != 0:
```

Comment:
I think I tried to studiously avoid the use of `~` because with unbounded integers (python integers), because I don't know/trust what it'll do when applied to a fixed bit number, although apparently I've let it sneak in three times already...  5:S  As long as one of the two number is bounded, it should be fine because it's just being applied as a mask... so no way to create a number bigger than the minimum of the two...  I just worry about it extending far enough to cover large numbers, but I suppose python's already figured that out...

Reply:
yeah, we have to be careful with that and I found some issues already.. but that's safe because it has the AND with the integer (page_address)